cmd/docker-scan/main.go

```diff
@@ -157,6 +157,9 @@ func checkConsent(flags options, dockerCli command.Streams) (config.Config, erro
 	if err != nil {
 		return config.Config{}, err
 	}
+	if flags.showVersion {
```

Comment:
Can we move this even further up? Looks like now it first reads the config file (which I don't think is needed to show --version?) https://github.com/docker/scan-cli-plugin/blob/main/cmd/docker-scan/main.go#L156

Reply:
It's actually needed, on Desktop it contains the path to Snyk binary, we invoke this to get the snyk version that is displayed

Reply:
Hmm.. that's a bit unexpected for a `--version` flag to execute some other binary.

Would that mean that on Linux (where I think it's ran in a container?) running `--version` would
- connect to the daemon
- start a container?

Reply:
Yes indeed that's what it does on linux 😅 
I would have prefer a `version` command instead, but the whole scan plugin is a single command, not a manage one...

Reply:
Definitely not something I would expect a `--version` flag to do; `--version` should just print the version of the binary itself, not actually run things. Do we absolutely need that information in `--version`? Think of situation where;

- `docker context` points to my production server, and I would use `--version`; in that case it would (`docker pull` and) `docker run` a container on that server 🙀 
- I have issues with my daemon; `--version` can be part of troubleshooting, and now won't be able to work, because it requires the daemon to run (and being able to connect)

Reply:
Hmm... I also see that 

- "loading config file" _creates_ directories and  _creates_ the config file
- looks like there's various references assuming Docker Desktop (which won't be the case on Linux)

https://github.com/docker/scan-cli-plugin/blob/cafb06dfc214a6e0bd31e71ea0732cf16ebaee3d/config/config.go#L45-L59

Reply:
Yes, that's what it does for now (unfortunately, I agree...)
To improve this we would need a way to have a comprehensive link between a snyk image and the binary version, that's another issue separated from this.